packages/core/src/new-api/internal/module-builder.ts

```diff
@@ -120,6 +120,12 @@ export class ModuleConstructor {
       )
     );
 
+    if ((mod as any).results instanceof Promise) {
+      throw new IgnitionError(
```

Comment:
This was feedback from Argentina's hackathon

Reply:
nice!